validator/src/main.rs

```diff
@@ -2706,44 +2706,45 @@ pub fn main() {
         validator_config.max_ledger_shreds = Some(limit_ledger_size);
     }
 
-    validator_config.ledger_column_options = LedgerColumnOptions {
-        compression_type: match matches.value_of("rocksdb_ledger_compression") {
-            None => BlockstoreCompressionType::default(),
-            Some(ledger_compression_string) => match ledger_compression_string {
-                "none" => BlockstoreCompressionType::None,
-                "snappy" => BlockstoreCompressionType::Snappy,
-                "lz4" => BlockstoreCompressionType::Lz4,
-                "zlib" => BlockstoreCompressionType::Zlib,
-                _ => panic!(
-                    "Unsupported ledger_compression: {}",
-                    ledger_compression_string
-                ),
-            },
+    let rocks_compression_type = match matches.value_of("rocksdb_ledger_compression") {
```

Comment:
Might be better to move all these code and below to a helper function in `validator/src/main.rs`;
Something like:
```rust
fn make_ledger_column_options(matches: &ArgMatches<'_>) -> LedgerColumnOptions {
   ...
}
```
and then in `fn main()` just do
```rust
validator_config.ledger_column_options = make_ledger_column_options(&matches);
```